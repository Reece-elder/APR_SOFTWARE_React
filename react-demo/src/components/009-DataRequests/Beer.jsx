const Beer = (props) => {

    const{name, image, abv, foodPairing} = props; 

    return ( 
        <div>
            <h3> Beer name : {name}</h3>
            <img src={image} width={200} alt={name}/>
            <h4> Beer abv : {abv}</h4>
            {foodPairing.map((x) => (
                <p> Food Pairing: {x}</p>
            ))}
        </div>
     );
}
 
export default Beer;