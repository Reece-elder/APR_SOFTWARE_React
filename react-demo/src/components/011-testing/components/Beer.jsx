const Beer = (props) => {
    const{name, image, abv, foodPairing} = props;

    return ( 
        <div>
            <h3>{name}</h3>
            <img src={image} alt={name} width={70}></img>
            <p>Alcohol content of beer is {abv}</p>
            <h4> Good food pairings with this are: </h4>
            {foodPairing.map((food) => (
                <p>{food}</p>
            ))}
        </div>
     );
}
 
export default Beer;