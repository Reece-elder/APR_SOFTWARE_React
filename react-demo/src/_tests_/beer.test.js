import Beer from "../components/011-testing/components/Beer";

import {create} from 'react-test-renderer'

describe(`Beer component testing with props`, () => {

    // Variable for the props to pass into beer
    const beerData = {
        name: 'test_name',
        image: 'test_url',
        abv: 123,
        foodPairing: ['food_pairing1', 'food_pairing2']
    };

    // Create a variable called testComponent
    let testComponent;

    // before each test create a new component and assign to test component
    beforeEach(() => {
        const TestInstance = create(<Beer {...beerData}/>);
        testComponent = TestInstance.root;
    });

    it(`a should be equal to 5`, () => {
        const a = 5;
        expect(a).toBe(5);
    });

    it(`Should render a h3 which contains the name of the beer`, () => {
        // Arrange + Act
        // Create h3 variable, it is equal to the first h3 in our testComponent
        const h3 = testComponent.findByType('h3');

        // Assert
        // Expecting to recieve an [array] so to equal must be an array
        expect(h3.children).toEqual([beerData.name]);
    });

    it(`Should render the FIRST p which should contain the abv`, () => {
        const firstP = testComponent.findAllByType('p')[0];
        expect(firstP.children[1]).toContain([beerData.abv]);
    });

    it(`Should render a h4 which contains the string 'Good food pairings..`, () => {
        const h4 = testComponent.findByType('h4');
        expect(h4.children.toString()).toMatch('Good food pairings');
    })

});

describe(`Snapshot testing`,()=> {

        // Variable for the props to pass into beer
        const beerData = {
            name: 'test_name',
            image: 'test_url',
            abv: 123,
            foodPairing: ['food_pairing1', 'food_pairing2']
        };
    
        // Create a variable called testComponent
        let testComponent;
    
        // before each test create a new component and assign to test component
        beforeEach(() => {
            const TestInstance = create(<Beer {...beerData}/>);
            testComponent = TestInstance.root;
        });

        // Snapshot test - Takes a snapshot of the code (in JSON), checks the current code against the snapshot
        // If match - Test passes
        // If not match.. test fails. 

        it(`Should match the snapshot`, () => {
            const element = create(<Beer {...beerData}/>).toJSON();
            expect(element).toMatchSnapshot();
        })

        it(`a = 5`, () => {
            const a = 5;
            expect(a).toBe(5);
        })
    

})