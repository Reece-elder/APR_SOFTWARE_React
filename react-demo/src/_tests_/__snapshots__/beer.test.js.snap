// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Should match the snapshot 1`] = `
<div>
  <h3>
    test_name
  </h3>
  <img
    alt="test_name"
    src="test_url"
    width={70}
  />
  <p>
    Alcohol content of beer is 
    123
  </p>
  <h4>
    Good food pairings with this are: 
  </h4>
  <p>
    food_pairing1
  </p>
  <p>
    food_pairing2
  </p>
</div>
`;
